server/api/setUserPuzzleMeta.js

```diff
@@ -0,0 +1,77 @@
+// Other UserPuzzle relationship routes will be placed here.
+// Eventually, the `time` metadata might be set here as well.
+const { db } = require("../db");
+
+/**
+ * The callback function for the /api/userPuzzleMeta POST route.
+ * Sets the value of the completion of a puzzle (i.e. UserPuzzle.progress).
+ *
+ * @param {Express.Request} req - The Request object generated by the route.
+ *      Should contain the following keys:
+ *          - `progress` - The decimal representation of the binary number defining the progress of the puzzle.
+ *              For now, the number will be `0` for an incomplete puzzle and `1` for a complete puzzle.
+ *          - `puzzle_id` - The `puzzle_id` value of the Puzzle with which to update the UserPuzzle relation.
+ *          - `time` - The integer time (in milliseconds) in which the user finished the puzzle.
+ * @param {Express.Response} res - The Result object generated by the route.
+ * @param {Function} next - The next() function used in the callback.
+ */
+function setUserPuzzleMeta(req, res, next) {
+  // safety net
+  if (
+    !("progress" in req.body && "puzzle_id" in req.body && "time" in req.body)
```

Comment:
`!req.body.progress || !req.body.puzzle_id || !req.body.time` or 
`!(req.body.progress && req.body.puzzle_id && req.body.time)` should work if you prefer

Reply:
I would also add that all 3 parameters should be validated as numbers if possible. Either with `!isNaN(value)` or some quick regex

Reply:
Did the second thing, but the first will fail if `progress` or `time` are 0 (since 0 is falsy); so, I'm not gonna do it